### ISSUE TITLE
(Sighting) Spawner Break-ability and other Mods

### ISSUE THREAD

**Issue by t3f** [PrinceOfAmber/Cyclic#56]
I've not isolated what mod(s) Cyclic is interacting with yet, but I wanted to report this in case someone else has seen it and maybe knows the answer.  

I've observed a behavior where when in the config "Spawners Unbreakable"=false AND some other mod is installed Spawners are yet still unbreakable.  Now I've been doing some fiddling and a fresh world with _just_ cyclic and a minecraft:mob_spawner obeys the setting just fine and is breakable.  Also with the other mods loaded on a a fresh world, spawners are again breakable.  

What i'm seeing be unpredictable is when I start adding Cyclic back to a world that had other mods.  In single player, in existing chunks, it looks like if the spawner already existed that I can still break it, but in newly generated chunks that may not always be the case.  

Again, still need to do some A|B testing to see if I can isolate this.  This was encountered on a server where we had roguelike dungeons installed and later added Cyclic and we observed spawners were suddenly invincible.

**Comment by PrinceOfAmber** [comment]
Ill do some testing too. I only tried in single player, but changing that flag and restarting seemed to make them breakable again.  I didnt think pregenerated chunks should matter.

Literally this is the code

if (spawnersUnbreakable) {
      Blocks.MOB_SPAWNER.setBlockUnbreakable();
    }

That config just sets a property of the block when the game/server starts.  Anyway thanks for the report ill look into it some more.

**Comment by PrinceOfAmber** [comment]
https://github.com/PrinceOfAmber/Cyclic/blob/bcec21a5299a83ebb4bc0200afc9e0458d7d387c/src/main/java/com/lothrazar/cyclicmagic/registry/BlockRegistry.java#L69

**Comment by PrinceOfAmber** [comment]
Try this in your world when 1.4.2 download is out, or let me know what your worlds are doing with it.. Ill keep the issue open

**Comment by PrinceOfAmber** [comment]
https://github.com/PrinceOfAmber/Cyclic/blob/daa877535ab2d5d54b65dffd18f406f3d0075a72/src/main/java/com/lothrazar/cyclicmagic/registry/BlockRegistry.java#L72

**Comment by PrinceOfAmber** [comment]
sorry try it now http://minecraft.curseforge.com/projects/cyclic/files/2317553

**Comment by t3f** [comment]
Ah ha. At first I was going to say it didn't change the behavior, but I think I found the culprit combo as I started to iterate through removing mods.

It appears something about Cyclic and Bloodmagic don't play well.  The default config for Bloodmagic contains a blacklist for one of their functions (teleposer) that includes spawners, and when Cyclic is installed it seems to prevent destruction.  I removed the Spawner from their blacklist and the problem goes away with 1.4.2.  Without Cyclic installed, the unbreakable behavior doesnt exist.


For completeness, before I found that setting, what I what I tested was:
* Exiting world + existing mods - upgraded from Cyclic 1.4.0/.1 to .2  Keeping old Config File  
 * Couldn't break in survival
* Existing world + existing mods - with Cyclic .2 and regenerated config file and B:"Spawners Unbreakable"=false  
  * Couldn't break in survival
* New World + existing mod suite and Cyclic 1.4.2 
 * Couldn't Break in survival
* New World + existing mod suite and without Cyclic
 * Can Break in survival
* New World and only Cyclic
 * Can Break in survival

**Comment by PrinceOfAmber** [comment]
Thanks for all the details.  So with only cyclic the problem goes away, but somehow comes back in the modpack.

"New World + existing mod suite and Cyclic 1.4.2
Couldn't Break in survival"  

The new fix sets the hardness back to 5 from unbreakable as long as the config is false, so i wonder what else is going on there.   Ill leave the issue open for now, even though it sounds like some sort of mod conflict.

If you want an easy way to share out lists of mods in your world without making a pack, theres a mod called modlister.  I just install it and then you have a link like this https://modlister.com/user/Lothrazar

**Comment by PrinceOfAmber** [comment]
I made a non public test build that is only for this issue, to see if it fixes anything for you.  I tried to make it tie tighter to the config change to  undo the unbreakable change but im not having your exact situation.  Anyway, just try it if you want and let me know if it works.  

https://github.com/PrinceOfAmber/Cyclic/releases/tag/1.4.4prerelease

**Comment by PrinceOfAmber** [comment]
Its been 11 days, so ill assume that either the changes fixed things, or the issue is just some unexpected mod conflict and im not sure where to go from there.  Ill mark it as wontfix for now but the comments should stay open

**Comment by t3f** [comment]
Sorry, got rather busy/swamped.
I've not had a chance to regress with your pre-release build, but I can confirm that with 1.4.4 _and_ the Bloodmagic config work-around the behavior went away.  Interestingly enough I needed to also push the 1.4.4 config to the clients config too, anyone who had an old config saw this.  I agree that it's likely still closed, i'll try to come back around to this in the weekend regardless.  Thanks for your continued help and patience.

**Comment by PrinceOfAmber** [comment]
UPDATE: with the help of some other testers and a particular streamer, using the 'We Gonna Die' modpack which has roguelike dungeons, we figured this out.

They had unbreakable spawners, and got them to be breakable with one important caveat:

You must turn the config to false on both the Server AND the client.  After that, i saw live on stream that they were breakable again.

Anyway I know this has been closed for a while but I wanted you to know.
